// weather-man/utils/constants.js
module.exports = {
    FAHRENHEIT: 'f',
    CELCIUS: 'c',
    KELVIN: 'k',

    KILOMETERS: 'km/h',
    MILES: 'mi/h',
    METERS: 'm/s',

    AMPM: 'am/pm',

    YAHOO: 'yahoo',
    OPENWEATHERMAP: 'openweathermap',
    YRNO: 'yrno',
    FORECASTIO: 'forecastio',
    DARKSKY: 'darksky',
    AQICN: 'aqicn',

    CLEAR: 'clear',
    CLOUDY: 'cloudy',
    FOG: 'fog',
    LIGHT_RAIN: 'light rain',
    RAIN: 'rain',
    THUNDERSTORM: 'thunderstorm',
    SNOW: 'snow',
    HAIL: 'hail',
    WIND: 'wind',
    EXTREME_WIND: 'extreme wind',
    TORNADO: 'tornado',
    HURRICANE: 'hurricane',
    EXTREME_COLD: 'extreme cold',
    EXTREME_HEAT: 'extreme heat',
    SNOW_THUNDERSTORM: 'snow thunderstorm',

// flat result object keys

    LAT: 'lat',
    LON: 'lon',
    TEMP: 'temp',
    MAX: 'max',
    MIN: 'min',
    CODE: 'code',
    SUMMARY: 'summary',
    HUMIDITY: 'humidity',
    PRESSURE: 'pressure',
    VISIBILITY: 'visibility',
    SUNRISE: 'sunrise',
    SUNSET: 'sunset',
    WIND_SPEED: 'windSpeed',
    WIND_DIR: 'windDir',
    FEELS_LIKE: 'feelsLike'
};
